pkg/build/builder/dockerutil.go

```diff
@@ -62,15 +73,35 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	if glog.V(5) {
 		opts.OutputStream = os.Stderr
 	}
-	var err error
+
 	var retriableError = false
 
+	eventWatch, err := eventClient.Watch(
```

Comment:
no, just list the events with the same filter, and use the resourceVersion on the resulting list to start your watch